packages/react-core/src/demos/ComposableMenu/examples/ComposableMultipleTypeaheadSelect.tsx

```diff
@@ -0,0 +1,219 @@
+          indexToFocus = menuItems.length - 1;
+        } else {
+          indexToFocus = focusedItemIndex - 1;
+        }
+      }
+
+      if (key === 'ArrowDown') {
+        // When no index is set or at the last index, focus to the first, otherwise increment focus index
+        if (focusedItemIndex === null || focusedItemIndex === menuItems.length - 1) {
+          indexToFocus = 0;
+        } else {
+          indexToFocus = focusedItemIndex + 1;
+        }
+      }
+
+      setFocusedItemIndex(indexToFocus);
+    }
+  };
+
+  const onInputKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    const enabledMenuItems = menuItems.filter(menuItem => !menuItem.isDisabled);
+    const [firstMenuItem] = enabledMenuItems;
+    const focusedItem = focusedItemIndex ? enabledMenuItems[focusedItemIndex] : firstMenuItem;
+
+    switch (event.key) {
+      // Select the first available option
+      case 'Enter':
+        if (!isMenuOpen) {
+          setIsMenuOpen(prevIsOpen => !prevIsOpen);
+        } else {
+          onMenuSelect(focusedItem.itemId as string);
+        }
+        break;
+      case 'Tab':
+      case 'Escape':
+        setIsMenuOpen(false);
+        break;
+      case 'ArrowUp':
+      case 'ArrowDown':
+        handleMenuArrowKeys(event.key);
+        break;
+      default:
+        !isMenuOpen && setIsMenuOpen(true);
+    }
+  };
+
+  // Close the menu when a click occurs outside of the menu toggle content
+  const onDocumentClick = (event: MouseEvent) => {
+    if (
+      isMenuOpen &&
+      !menuToggleRef.current?.contains(event?.target as HTMLElement) &&
+      !menuRef?.current?.contains(event.target as Node)
+    ) {
+      setIsMenuOpen(false);
+    }
+  };
+
+  const toggleMenuOpen = () => {
+    setIsMenuOpen(!isMenuOpen);
+    textInputRef.current?.focus();
+  };
+
+  const onTextInputChange = (value: string) => {
+    setInputValue(value);
+  };
+
+  const onMenuSelect = (itemId: string) => {
+    if (itemId) {
+      setSelected(
+        selected.includes(itemId) ? selected.filter(selection => selection !== itemId) : [...selected, itemId]
+      );
+    }
+  };
+
+  const toggle = (
+    <MenuToggle
+      variant="typeahead"
+      onClick={toggleMenuOpen}
+      innerRef={menuToggleRef}
+      isExpanded={isMenuOpen}
+      isFullWidth
+    >
+      <TextInputGroup isPlain>
+        <ChipGroup>
+          {selected.map((selection, index) => (
+            <Chip
+              key={index}
+              onClick={ev => {
+                ev.stopPropagation();
+                onMenuSelect(selection);
+              }}
+            >
+              {selection}
+            </Chip>
+          ))}
+        </ChipGroup>
+        <TextInputGroupMain
+          value={inputValue}
+          onClick={toggleMenuOpen}
+          onChange={onTextInputChange}
+          onKeyDown={onInputKeyDown}
+          id="typeahead-select-input"
+          autoComplete="off"
+          innerRef={textInputRef}
+        />
```

Comment:
Getting some left-edge-no-space action with the chip group:

<img width="288" alt="Screen Shot 2022-09-07 at 6 13 40 PM" src="https://user-images.githubusercontent.com/35148959/188999197-cd4c51e4-7cef-48f5-9b65-16a61d634986.png">

From the [text input group filters example](https://www.patternfly.org/v4/components/text-input-group#filters), looks like `<ChipGroup />` needs to be a child of `<TextInputGroupMain>`

```suggestion
        <TextInputGroupMain
          value={inputValue}
          onClick={toggleMenuOpen}
          onChange={onTextInputChange}
          onKeyDown={onInputKeyDown}
          id="typeahead-select-input"
          autoComplete="off"
          innerRef={textInputRef}
        >
          <ChipGroup>
            {selected.map((selection, index) => (
              <Chip
                key={index}
                onClick={ev => {
                  ev.stopPropagation();
                  onMenuSelect(selection);
                }}
              >
                {selection}
              </Chip>
            ))}
          </ChipGroup>
        </TextInputGroupMain>
```